libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -85,6 +85,11 @@ public SpanScope withSpanInScope(Span span) {
         return DefaultSpanScope.create(span, tracerContextStorage).attach();
     }
 
+    @Override
+    public boolean isRecording() {
+        return true;
```

Comment:
I think this guy should consult settings & feature flag?

Reply:
Ideally, that all should be take care before. But we can add that check here as well.

Reply:
Got it, I agree: since flags are static, no need to recheck them every time.